src/server/crm/ongage.js

```diff
@@ -0,0 +1,216 @@
+//@flow
+
+import fetch from 'cross-fetch'
+import { get, isNil, fromPairs } from 'lodash'
+import { UserRecord } from '../../imports/types'
+import logger from '../../imports/logger'
+import Config from '../server.config'
+import requestTimeout from '../utils/timeout'
+
+export type Contact = {
+  identifier: string,
+  first_name: string,
+  last_name: string,
+  mobile?: string,
+  email: string,
+  regmethod: string,
+  torusprovider: string,
+  term_utm: string,
+  content_utm: string,
+  source_utm: string,
+  medium_utm: string,
+  campaign_utm: string,
+  whitelisted?: string,
+  version_joined?: string,
+  signup_completed?: string
+}
+
+interface CRMAPI {
+  createContact(contact: UserRecord, logger): string;
+  updateContact(identifier: string, fields: { [key: string]: stirng }, logger): string;
+  userRecordToContact(contact: UserRecord): Contact;
+  deleteContactFromDNC(email: string, logger): any;
+  addContactToDNC(email: string, logger): any;
+  getContactByEmail(email: string, logger): any;
+  getContactById(id: string, logger): any;
+  deleteContact(id: string, logger): any;
+  setWhitelisted(id: string, logger): any;
+}
+
+const tagsMap = {
+  utmctr: 'term_utm',
+  utmcct: 'content_utm',
+  utmcsr: 'source_utm',
+  utmcmd: 'medium_utm',
+  utmccn: 'campaign_utm'
+}
+
+const fieldsMap = {
+  firstName: 'first_name',
+  lastName: 'last_name',
+  fullName: 'fullname',
+  regMethod: 'regmethod',
+  torusProvider: 'torusprovider'
+}
+
+export const parseUtmString = utmString => {
+  return (utmString || '').split('|').reduce((tags, record) => {
+    const [name, value] = record.split('=')
+    const tagValue = decodeURIComponent(value)
+
+    if (name in tagsMap && tagValue && '(not set)' !== tagValue) {
+      const mappedName = tagsMap[name]
+
+      tags[mappedName] = tagValue
+    }
+
+    return tags
+  }, {})
+}
+
+export class OnGage implements CRMAPI {
+  baseUrl: string
+  log = logger.child({ from: 'OnGage' })
+
+  constructor(apiurl: string, account_code: string, apikey: string, apisecret: string) {
+    this.baseUrl = apiurl
+    this.baseHeaders = {
+      'Content-Type': 'application/json',
+      X_USERNAME: apikey,
+      X_PASSWORD: apisecret,
+      X_ACCOUNT_CODE: account_code
+    }
+  }
+
+  baseQuery(url, headers, body = null, method = 'post', timeout = 15000) {
+    const { baseUrl, log } = this
```

Comment:
if you will use axios, you won't need this method.
you could create axios intsance and configure all this stuff with request / response / error handlers
then just call .get / .post etc of `this.axios`

Reply:
done